scripts/mbedtls_dev/crypto_reference.py

```diff
@@ -0,0 +1,238 @@
+"""Reference implementation of some cryptographic mechanisms.
+
+This module can be used to generate test cases.
+
+This module requires the Cryptodome library installed with
+``pip install pycryptodomex``. You need at least version 3.7.3 for
+the type subs for mypy.
+"""
+
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#pylint: disable=fixme,unused-import
+
+import enum
+import types
+import typing
+from typing import Dict, List, Optional, Set, Tuple
+
+# TODO: support pycryptodome, which installs the Cryptodome library under the
+# module name ``Crypto``.
+from Cryptodome import Hash, Protocol, PublicKey
+import Cryptodome.Hash.MD5
```

Comment:
You're referring to these as e.g. "Hash.MD5" everywhere so no need for the individual imports

Reply:
They are needed actually.
```
~ % python3
Python 3.8.10 (default, Sep 28 2021, 16:10:42) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from Cryptodome import Hash
>>> Hash.MD5
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'Cryptodome.Hash' has no attribute 'MD5'
```
If you know why I'm all ears.